pkg/k8sapi/discovery.go

```diff
@@ -208,7 +208,7 @@ func (d *Controller) handlePodUpdate(key string) error {
 	}
 
 	pod, ok := obj.(*v1.Pod)
-	if !ok {
+	if !ok || pod == nil {
```

Comment:
What circumstances cause pod to be nil?

Reply:
@nckturner it happens once in @rtripat setup once.  I am not able to reproduce it.

Reply:
Ok well it looks to me like it might help prevent #168 but I'm wondering why its receiving `nil` in the first place.

Reply:
@ofiliz 

I think this is OK.
A nil interface is != nil but a nil interface converted to  struct is nil and pod is a struct.

https://play.golang.org/p/uAZJG2_mUuL

Unless I am missing something.

Reply:
lgtm

I agree with nic that we should investigate further why we can get a nil pod here if exists is true.  Somewhere we are storing nil in the datastore.